contracts/eosio.system/src/rex.cpp

```diff
@@ -664,7 +664,25 @@ namespace eosiosystem {
 
          if ( new_return_bucket ) {
             _rexretbuckets.modify( ret_buckets_elem, same_payer, [&]( auto& rb ) {
-               rb.return_buckets[new_bucket_time] = new_bucket_rate;
+               auto& return_buckets = rb.return_buckets;
+               auto iter = return_buckets.begin();
+               auto added = false;
+               // sort by first
+               while ( iter != return_buckets.end() ) {
+                  if ( new_bucket_time < iter->first ) {
+                     return_buckets.insert( iter, pair_time_point_sec_int64 { new_bucket_time, new_bucket_rate });
+                     added = true;
+                     break;
+                  } else if ( new_bucket_time == iter->first ) {
+                     *iter = pair_time_point_sec_int64 { new_bucket_time, new_bucket_rate };
+                     added = true;
+                     break;
+                  }
+                  ++iter;
+               }
+               if ( !added ) {
+                  return_buckets.emplace_back( pair_time_point_sec_int64 { new_bucket_time, new_bucket_rate });
+               }
```

Comment:
This does not appear to keep the buckets in `rb.return_buckets` sorted by `first`.

Instead replace with the follow which does maintain sort order and is simpler:
```cpp
auto iter = std::lower_bound(rb.return_buckets.begin(), rb.return_buckets.end());
if ((iter != return_buckets.end()) && (iter->first == new_bucket_time)) {
   iter->second = new_bucket_rate;
} else {
   rb.return_buckets.insert(iter, pair_time_point_sec_int64{new_bucket_time, new_bucket_rate});
}
```

Reply:
Thanks. Your way is cleaner and simpler. Will use it.